internal/crypto/src/cose/certificate_trust_policy.rs

```diff
@@ -54,21 +57,67 @@ impl Default for CertificateAcceptancePolicy {
     }
 }
 
-impl CertificateAcceptancePolicy {
+impl CertificateTrustPolicy {
     /// Create a new certificate acceptance policy with no preconfigured trust
     /// roots.
     ///
     /// Use [`default()`] if you want a typical built-in configuration.
     ///
     /// [`default()`]: Self::default()
     pub fn new() -> Self {
-        CertificateAcceptancePolicy {
+        CertificateTrustPolicy {
             trust_anchor_ders: vec![],
             end_entity_cert_ders: vec![],
             additional_ekus: HashSet::default(),
         }
     }
 
+    /// Evaluate a certificate against the trust policy described by this
+    /// struct.
+    ///
+    /// Returns `Ok(())` if the certificate appears on the end-entity
+    /// certificate list or has a valid chain to one of the trust anchors that
+    /// was provided and that it has a valid extended key usage (EKU).
+    ///
+    /// If `signing_time_epoch` is provided, evaluates the signing time (which
+    /// must be in Unix seconds since the epoch) against the certificate's
+    /// period of validity.
+    #[allow(unused)] // parameters may be unused in some cases
+    #[async_generic]
+    pub fn check_certificate_trust(
+        &self,
+        chain_der: &[Vec<u8>],
+        end_entity_cert_der: &[u8],
+        signing_time_epoch: Option<i64>,
+    ) -> Result<(), CertificateTrustError> {
+        if _async {
```

Comment:
Do we support the sync wasm path now?

Reply:
No. The non `_async` case will fall through to the error response at line 116.

Reply:
Ok, a todo for Colin's case.